src/components/fields/text-field/text-field.tsx

```diff
@@ -110,8 +110,8 @@ export const TextField = (props: IGenericFieldProps<ITextFieldSchema | IEmailFie
 	// EVENT HANDLERS
 	// =============================================================================
 	const handleChange = (event: React.ChangeEvent<HTMLInputElement>): void => {
-		if (uiType === "numeric-field" && !event.target.value) {
-			onChange({ target: { value: undefined } });
+		if (uiType === "numeric-field") {
+			onChange({ target: { value: +event.target.value ?? undefined } });
```

Comment:
`+""` becomes 0 which may not be intended. we may still need the falsy conditon to set undefined, but input to be cast to number